pkg/cmd/release/update/update.go

```diff
@@ -0,0 +1,186 @@
+			$ gh release update 63899317 --prerelease=false
+
+			Update the target commitish of a release
+			$ gh release update 63899317 --target 97ea5e77b4d61d5d80ed08f7512847dee3ec9af5
+
+			Update the name of the tag
+			$ gh release update 63899317 --tag v9.8.7
+
+			Update the name/title of a release
+			$ gh release update 63899317 --title 'Some new title'
+
+			Update the release notes
+			$ gh release update 63899317 --notes 'Some short release notes'
+
+			Update the release notes from the content of a file
+			$ gh release update 63899317 --notes-file /path/to/release_notes.md
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.ReleaseId = args[0]
+
+			opts.BaseRepo = f.BaseRepo
+			opts.RepoOverride, _ = cmd.Flags().GetString("repo")
+
+			err := parseOptionalBoolFlags(cmd, opts)
+			if err != nil {
+				return err
+			}
+
+			opts.BodyProvided = cmd.Flags().Changed("notes")
+			if notesFile != "" {
+				b, err := cmdutil.ReadFile(notesFile, opts.IO.In)
+				if err != nil {
+					return err
+				}
+				opts.Body = string(b)
+				opts.BodyProvided = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return updateRun(opts)
+		},
+	}
+
+	cmd.Flags().Bool("draft", false, "Save the release as a draft instead of publishing it")
+	cmd.Flags().Bool("publish", false, "Publish the release")
+	cmd.Flags().Bool("prerelease", false, "Mark the release as a prerelease")
+	cmd.Flags().StringVar(&opts.Target, "target", "", "Target `branch` or full commit SHA (default: main branch)")
+	cmd.Flags().StringVar(&opts.TagName, "tag", "", "The name of the tag")
+	cmd.Flags().StringVarP(&opts.Name, "title", "t", "", "Release title")
+	cmd.Flags().StringVarP(&opts.Body, "notes", "n", "", "Release notes")
+	cmd.Flags().StringVarP(&notesFile, "notes-file", "F", "", "Read release notes from `file` (use \"-\" to read from standard input)")
+
+	return cmd
+}
+
+func updateRun(opts *UpdateOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	params := getParams(opts)
+
+	if len(params) == 0 {
+		return errors.New("nothing to update")
+	}
+
+	updatedRelease, err := updateRelease(httpClient, baseRepo, opts.ReleaseId, params)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(opts.IO.Out, "%s\n", updatedRelease.URL)
+
+	return nil
+}
+
+func parseOptionalBoolFlags(cmd *cobra.Command, opts *UpdateOptions) error {
+	if cmd.Flags().Changed("draft") {
+		draft, err := cmd.Flags().GetBool("draft")
+		if err != nil {
+			return err
+		}
+		opts.Draft = &draft
+	}
+
+	if cmd.Flags().Changed("prerelease") {
+		prerelease, err := cmd.Flags().GetBool("prerelease")
+		if err != nil {
+			return err
+		}
+		opts.Prerelease = &prerelease
+	}
+
+	return nil
+}
+
+func getParams(opts *UpdateOptions) map[string]interface{} {
+	params := map[string]interface{}{}
+
+	if opts.Body != "" {
```

Comment:
If a blank release body is allowed by the platform, we should allow the update command to explicitly set the body to an empty string. This condition prevents that.

To differentiate between an explicitly blank body set by the user vs. body hasn't been defined by the user, you can use `cmdutil.NilStringFlag()` and connect it to a string pointer.

```go
type Opts struct {
  Body *string
}

cmdutil.NilStringFlag(cmd, &opts.Body, "--notes", "-n", "Set release notes")
```